src/coreclr/tools/Common/Compiler/DependencyAnalysis/Relocation.cs

```diff
@@ -549,6 +588,9 @@ public static unsafe void WriteValue(RelocType relocType, void* location, long v
                 case RelocType.IMAGE_REL_BASED_RISCV64_PC:
                     PutRiscV64PC((uint*)location, value);
                     break;
+                case RelocType.IMAGE_REL_BASED_RISCV64_JALR:
+                    PutRiscV64JALR((uint*)location, value);
+                    break;
```

Comment:
```suggestion
                case RelocType.IMAGE_REL_BASED_RISCV64_PC:
                case RelocType.IMAGE_REL_BASED_RISCV64_JALR:
                    PutRiscV64PC((uint*)location, value);
                    break;
```
I guess now they are the same; the immediates of load, jalr, and addi have the same bit layout.

You can just add a tighter assertion for 2nd instruction in IMAGE_REL_BASED_RISCV64_JALR.